tests/test_download_orchestrator.py

```
@@ -3284,6 +4337,110 @@ def test_process_firmware_uses_latest_by_version_with_mixed_releases(
+        orch.firmware_downloader.should_download_release = Mock(return_value=True)
+        orch.firmware_downloader.download_repo_prerelease_firmware = Mock(
+            return_value=([], [], None, None)
+        )
+        orch.firmware_downloader.collect_non_revoked_releases = Mock(
+            return_value=([newer, older], [newer, older], 8)
+        )
+        orch.firmware_downloader.is_release_revoked = Mock(return_value=False)
+
+        side_effects = {id(newer): False, id(older): True}
+
+        def _download_firmware_side_effect(release):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_download_firmware_side_effect`

(ANN202)